File: cpp/src/exampleC.cpp


#include "exampleC.h"

using namespace exampleC;

exampleC::ItemBuilder::ItemBuilder(string name){

}
ItemBuilder exampleC::ItemBuilder::withPrice(double price){
    throw "not implemented";
}
ItemBuilder exampleC::ItemBuilder::withQuantity(double quantity){
    throw "not implemented";
}
Item exampleC::ItemBuilder::build(){
    throw "not implemented";
}

double exampleC::ShoppingBasket::getQuantity(string name){
    throw "not implemented";

}
double exampleC::ShoppingBasket::calculateTotal(){
    throw "not implemented";
}

ShoppingBasketBuilder exampleC::ShoppingBasketBuilder::addItem(Item item){
    throw "not implemented";

}
ShoppingBasket exampleC::ShoppingBasketBuilder::build(){
    throw "not implemented";

}

File: typescript/src/exampleC.ts
export class ShoppingBasket {
  getQuantity(itemName: String): number {
    throw new Error("Not implemented yet");
  }

  totalPrice(): number {
    throw new Error("Not implemented yet");
  }
}
export class Item {}

export class ShoppingBasketBuilder {
  addItem(item: Item): ShoppingBasketBuilder {
    throw new Error("Not implemented yet");
  }

  build(): ShoppingBasket {
    throw new Error("Not implemented yet");
  }
}

export class ItemBuilder {
  private name: string;

  constructor(name: string) {
    throw new Error("Not implemented yet");
  }

  withPrice(price: number): this {
    throw new Error("Not implemented yet");
  }

  withQuantity(quantity: number): this {
    throw new Error("Not implemented yet");
  }

  build(): Item {
    throw new Error("Not implemented yet");
  }
}
